### DISCUSSION TITLE
Any way to send variable to ahk code when starting script?

### DISCUSSION THREAD

**Discussion by Gh0sTG0** [spyoungtech/ahk#241] [Q&A] [answered]
Hi. Can I, somehow, start ahk code, but make it accept some variable as a part of the code?
Smth like, I have in my `some_loop.ahk` file:
```


Loop, %my_variable%
    MsgBox, %A_Index% | %my_variable%
ExitApp
```
And in my python code:
```
from ahk import AHK

ahk = AHK()
how_many_loops = 10
script_path = r'd:\some_loop.ahk'
ahk.run_script(script_path)
```

So, is there some way to transfer `how_many_loops` into ahk's `my_variable`?

Ofc I can do something like:
```
ahk_script1 = f'num = {how_many_loops}'
with open(r'd:\some_loop.ahk', 'r', -1, 'UTF-8') as file:
    ahk_script2 = file.read()

ahk_script = ahk_script1 + ahk_script2
ahk.run_script(ahk_script)
```
Plus add some lines to beginning of `some_loop.ahk`, `Biga` function, because for some reasons ahk stubbornly continues thinking that num is string and should be processed only as string. And no, it really don't want to use it as number, until Biga converts it...):
```


the_biga := new biga()
my_variable := the_biga.parseInt(num)

Loop, %my_variable%
    MsgBox, %A_Index% | %my_variable%
ExitApp
```
But maybe there are some better ways?

**Comment by spyoungtech** [comment, marked as answer]
I think you've got the right idea by modifying your script using Python to inject your variable before calling `run_script`.

Though, I'm not sure what your issue is with the use of your variable with your loop or why you need to use BigA.

For example, this works:

```autohotkey
num = 3

; This will loop 3 times
Loop, %num% {
    MsgBox, Loop %A_Index% out of %num%
}
```

Not sure if there's something I'm missing here.

**Reply by Gh0sTG0** [reply]
Oh, looks like I find what was the problem...
To make ahk understand words from not-english language I re-saved ahk script as `utf-8 with BOM`.
And, it somehow makes things broken...
If I re-save file as `utf-8` (without BOM) it starts understanding that `a:=5` is integer.
PS Looks like this should go to issues tab?
